tests/integration/cluster.go

```diff
@@ -1297,7 +1297,7 @@ func NewClusterV3(t testutil.TB, cfg *ClusterConfig) *ClusterV3 {
 	}
 	if !strings.HasPrefix(wd, os.TempDir()) {
 		t.Errorf("Working directory '%s' expected to be in temp-dir ('%s')."+
-			"Have you executed integration.BeforeTest(t) ?", wd, os.TempDir())
+			"Have you executed integration.RegisterLeakDetection(t) ?", wd, os.TempDir())
```

Comment:
the working directory is still set by BeforeTest

Reply:
Is this a problem after check being removed ?

The communication in etcd happens using local sockets that unfortunately needs to be in the current directory... and leaking tmp files in source directories is even more ugly.